### PR TITLE
Refactor parsing and serialisation code

### PR DESCRIPTION
**Pull request by flxo**
* Replace nom with byteorder::ReadBytesExt
    * Remove duplicate SomeIP prefix from types
    * Replace type tuple structs with typedefs for primitives
    * Adjust types::ReturnCode to standard definition (add reserved ranges)
    * Add Header::to_vec and Header::to_writer
    * Add SomeIp::tod_vec and SomeIp::to_writer
    * Add Header::from_slice and Header::from_reader
    * Add SomeIp::from_slice and SomeIp::from_reader
    * Deprecate serialize_someip_header in favor of Header::to_vec
    * Deprecate serialize_someip in favor of SomeIp::to_vec

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,13 +1,13 @@
 [package]
-name = "someip-parser"
-version = "0.2.0"
+name = "someip-header"
```
Comment (kruss) [inline Cargo.toml:2]: A naming like "someip-proto" or "someip-messages" would be more accurate for this crate, if it deals with the possible someip messages "on the wire" for SD and RPC (without the RPC specific types).

Comment (flxo) [reply Cargo.toml:2]: Agree. Possible more candidates:

* `someip-wire`
* `someip-basic`
* `someip-frame`

@marcmo @jens-esr ?

File: src/serializer.rs [hunk 2]
```diff
@@ -1,75 +1,112 @@
...
-    buf.put_u8(header.protocol_version);
-    buf.put_u8(header.interface_version);
-    buf.put_u8(header.message_type.into());
-    buf.put_u8(header.return_code.into());
+use byteorder::{BigEndian, WriteBytesExt};
+
+use crate::types::{Header, MessageID, MessageType, RequestID, ReturnCode, SomeIp};
+
+/// Header length in bytes PRS_SOMEIP_00030
+const HEADER_LENGTH: usize = 16;
```
Comment (kruss) [inline src/serializer.rs:8]: Duplicates HEADER_LENGTH in parser.rs

**Comment by kruss** [comment]
Nice cleanup! There are some clippy warnings left, but could be fixed in a later commit..

File: src/serializer.rs (hunk 2 above)
Comment (flxo) [reply src/serializer.rs:8]: Yes. Intentional because there's no better place for it. This number will never change.

File: Cargo.toml (hunk 1 above)
Comment (marcmo) [reply Cargo.toml:2]: I like `someip-message` or `someip-frame`

Comment (flxo) [reply Cargo.toml:2]: Everyone fine with `someip-message`? Think it's a good fit!

**Pull request merged**